sdk/python/feast/feature_service.py

```
@@ -85,32 +86,67 @@ def __init__(
-                            feature_grouping.projection.desired_features
-                        )
+                projection = feature_grouping.projection
+
+                if projection.desired_features:
+                    # The projection wants to select a specific set of inferred features.
+                    # Example: FeatureService(features=[fv[["inferred_feature"]]]), where
+                    # 'fv' is a feature view that was defined without a schema.
+                    if feature_grouping.name in fvs_to_update:
+                        # First we validate that the selected features have actually been inferred.
+                        desired_features = set(projection.desired_features)
                         actual_features = set(
                             [
                                 f.name
                                 for f in fvs_to_update[feature_grouping.name].features
                             ]
                         )
                         assert desired_features.issubset(actual_features)
-                        # We need to set the features for the projection at this point so we ensure we're starting with
-                        # an empty list.
-                        feature_grouping.projection.features = []
+
+                        # Then we extract the selected features and add them to the projection.
+                        projection.features = []
                         for f in fvs_to_update[feature_grouping.name].features:
                             if f.name in desired_features:
-                                feature_grouping.projection.features.append(f)
+                                projection.features.append(f)
                     else:
-                        feature_grouping.projection.features = fvs_to_update[
-                            feature_grouping.name
-                        ].features
+                        raise FeatureViewMissingDuringFeatureServiceInference(
```

probably should also throw this error in FeatureView#__getitem__ ?